### PR TITLE
Add support for TFUEL addresses

### LINKED ISSUE #159: Add support for Theta Fuel (TFUEL) addresses in encoding library
Here is the encoding library: https://github.com/ensdomains/address-encoder

Example: https://github.com/ensdomains/address-encoder/pull/47/files

Discord for dev help: https://discord.com/invite/AskZbFx

Please provide a reference link so we can check you did the encoding correctly.

We'd like to keep the encoding library as small as possible, so please be efficient in your use of space.

We will likely be reviewing PRs and paying out bounties more or less on a weekly basis. No need to ping us to check your PR, we will definitely get to it.

Thanks for your help!

### PR DESCRIPTION
**Pull request by Echolon166**
## Issue number
Closes #159 

## Description
Add support for TFUEL addresses, which uses the same address with Theta. 

## References
- https://github.com/thetatoken/theta-mainnet-integration-guide/tree/master/demos/keypair
- https://wallet.thetatoken.org/wallet/tokens/theta (TFUEL along with Theta in the same address)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My code implements all the required features.

### CODE REVIEW

**Comment by Echolon166** [comment]
Updated PR fixes the incorrect ordering in README.

**Comment by makoto** [comment]
@Echolon166 

Hi, I tried to resolve the conflict of this and the test failed again. Would you mind fixing this and also resolve conflicts of all the others you have submitted so that I can priortise your PR?

**Comment by Echolon166** [comment]
@makoto I've resolved the conflicts of my PR's.

File: src/__tests__/index.test.ts [hunk 1]
```diff
@@ -440,6 +440,13 @@ const vectors: Array<TestVector> = [
       { text: '6bNNutYQz11WrkVCrj1nUS1dBGyoVZjdEg', hex: 'a914e613c7be9b53e1a47fd4edb3ea9777cf29dce30f87' },
     ],
   },
+  {
+    name: 'TFUEL',
+    coinType: 570,
```
Comment (makoto) [inline src/__tests__/index.test.ts:445]: Is this correct coin type? 570 is allocated to BDX according to https://github.com/satoshilabs/slips/blob/master/slip-0044.md

File: src/index.ts [hunk 2]
```diff
@@ -679,6 +679,7 @@ export const formats: IFormat[] = [
   bitcoinBase58Chain('CCA', 489, [[0x0b]], [[0x05]]),
   getConfig('SOL', 501, bs58EncodeNoCheck, bs58DecodeNoCheck),
   bitcoinBase58Chain('LRG', 568, [[0x1e]], [[0x0d]]),
+  hexChecksumChain('TFUEL', 570),
```
Comment (makoto) [inline src/index.ts:682]: Is this correct coin type? 570 is allocated to BDX according to https://github.com/satoshilabs/slips/blob/master/slip-0044.md

**Comment by makoto** [comment]
Are there any block explorer for TFUEL? I want to double check that they are actually just using hex checksum

File: src/__tests__/index.test.ts (hunk 1 above)
Comment (Echolon166) [reply src/__tests__/index.test.ts:445]: I've opened a PR to add TFUEL to SLIP-0044: https://github.com/satoshilabs/slips/pull/1024
I will update this PR according to it when it finalizes.

**Comment by Echolon166** [comment]
TFUEL uses the same explorer with Theta token: https://explorer.thetatoken.org/

**Comment by Echolon166** [comment]
> Are there any block explorer for TFUEL? I want to double check that they are actually just using hex checksum

TFUEL uses the same explorer along with Theta token: https://explorer.thetatoken.org/

**Comment by makoto** [comment]
FYI, we have PR for THETA https://github.com/ensdomains/address-encoder/pull/125/files but implementation looks different. Are either of PR wrong or did the implementation change?

**Comment by Echolon166** [comment]
> FYI, we have PR for THETA https://github.com/ensdomains/address-encoder/pull/125/files but implementation looks different. Are either of PR wrong or did the implementation change?

We're both using the hexChecksumChain, so there's no practical difference. Seems like he just added some padding all around, for some reason.

I've updated the coin type.

**Pull request merged**